parquet/src/main/java/org/apache/iceberg/parquet/ParquetMetricsRowGroupFilter.java

```diff
@@ -580,6 +608,10 @@ static boolean hasNonNullButNoMinMax(Statistics statistics, long valueCount) {
         && (statistics.getMaxBytes() == null || statistics.getMinBytes() == null);
   }
 
+  static boolean minMaxUndefined(Statistics statistics) {
+    return !statistics.isEmpty() && !statistics.hasNonNullValue();
```

Comment:
This [unit test](https://github.com/apache/iceberg/issues/6516#issuecomment-1370862183) fails, I think it's because when all values are all null , min max will be null(undefined) too, seems we should handle the case of all values first.

Reply:
I am thinking we can simplify the handling of unreliable statistics by moving the judgment of all-null situation forward, as I mentioned [here](https://github.com/apache/iceberg/issues/6516#issuecomment-1370862183)，we can use `Statistics#getNumNulls() = ColumnChunkMetadata#getValueCount()` to determine if all values are null (which will not be affected by null or undinfined min max statistics); Then we handle the case of `statistics.hasNonNullValue()` is false(when statistics has null min max or unreliable min max), we should always return `ROWS_MIGHT_MATCH` when `statistics.hasNonNullValue()` is false bacause that means statistics is not reliable; Finally we can safely use min max for comparative evaluation.

Reply:
I think that sounds good, I was basing this determination off of the code here from Statistics.java in Parquet
```java
  @Override
  public String toString() {
    if (this.hasNonNullValue()) {
      if (isNumNullsSet()) {
        return String.format("min: %s, max: %s, num_nulls: %d", minAsString(), maxAsString(), this.getNumNulls());
      } else {
        return String.format("min: %s, max: %s, num_nulls not defined", minAsString(), maxAsString());
      }
    } else if (!this.isEmpty())
      return String.format("num_nulls: %d, min/max not defined", this.getNumNulls());
    else
      return "no stats for this column";
  }
  ```
  Which similarly checks for the numNulls stat first

Reply:
I simplified this a bit further since I don't think we have to handle the CDH special case in the same way. Now we basically just have a two step check

1) Are all the values null? This is only true if nullCount is set and it is equal to the value count from the chunk stats.

2) We check if min/max are defined for the stats
   a. For normal parquet stats this means that the stats are not empty && hasNonNull is false
   b. For the old CDH version this means the stats are not empty but min and max are set to null. I think the "a" case may also cover this based on my old notes but I figured since this check will be short circuited it doesn't hurt to leave it in.